tests/PHPStan/Analyser/data/substr-php8.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Substr;
+
+use function PHPStan\Testing\assertType;
+
+class Substr
+{
+	public function doFoo(string $s, int $i)
+	{
+		assertType('string', substr(null, true));
+		assertType('string', substr('a', 2));
```

Comment:
this cases always return an empty-string.. AFAIR there is no `empty-string` type, right?